daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -586,7 +617,7 @@ object Transaction {
         coinst: ContractInst[Value[TContractId]],
         optLocation: Option[Location],
         signatories: Set[Party],
-        stakeholders: Set[Party],
+        stakeholders: TreeSet[Party],
```

Comment:
why only stakeholders becomes `TreeSet`?

Reply:
I need the parties to be sorted because I added in the hash them in the `contractId`.

Reply:
Changing any `Set[Party]` to `TreeSet[Paty]`` in all the nodes implies a bit more work that I do not want to include in the PR. I changed back to Set and do the ordering just before hashing.